### ISSUE TITLE
Empty list/set results in distributed setup gives out an exception.

### ISSUE THREAD

**Issue by AkiRis** [orientechnologies/orientdb#5442]
OrientDB version : 2.1.7 distributed mode on two servers.

Following commands work correctly.

``` sql
SELECT null FROM V        /* result: field named "null" with value of null :D */
SELECT list(name) FROM V  /* name is existing field */
SELECT set(name) FROM V   /* name is existing field */
```

Following will give out Exceptions.
(note: following will work in non-distributed server)

``` sql
SELECT list(null) FROM V    /* not usefull, but should work */
SELECT set(null) FROM V     /* not usefull, but should work */
SELECT list(foobar) FROM V  /* foobar is non-existing field */
SELECT set(foobar) FROM V   /* foobar is non-existing field */
```

``` java
2015-12-09 15:30:42:596 SEVER Internal server error:
com.orientechnologies.orient.core.exception.OStorageException: Cannot route COMMAND operation to the distributed node
--> java.lang.IndexOutOfBoundsException: Index: 0, Size: 0 [ONetworkProtocolHttpDb]Cannot route COMMAND operation to the distributed node
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
    at java.util.ArrayList.rangeCheck(ArrayList.java:653)
    at java.util.ArrayList.get(ArrayList.java:429)
    at com.orientechnologies.orient.core.sql.functions.coll.OSQLFunctionList.mergeDistributedResult(OSQLFunctionList.java:95)
    at com.orientechnologies.orient.server.distributed.ODistributedStorage.mergeResultByAggregation(ODistributedStorage.java:449)
    at com.orientechnologies.orient.server.distributed.ODistributedStorage.command(ODistributedStorage.java:306)
    at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:67)
    at com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostCommand.execute(OServerCommandPostCommand.java:81)
    at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service(ONetworkProtocolHttpAbstract.java:180)
    at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute(ONetworkProtocolHttpAbstract.java:627)
    at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```

/ Aki R.

**Comment by AkiRis** [comment]
Any comments or estimates on timetable?

Bug prevents us from running local master of server,
so GUI client has to query data literally from other side of the globe.
(and lag is bad...)

**Comment by luigidellaquila** [comment]
Hi @AkiRis 

AFAIK it's already fixed in 2.2.x since a long time.

I'm closing this issue, please reopen if you still see the problem

Thanks

Luigi
